components/hab/src/license.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{common::ui::{self,
+                         UIReader,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            hcore::{fs::{am_i_root,
+                         FS_ROOT_PATH},
+                    users::get_current_username}};
+use chrono::prelude::*;
+use dirs;
+use serde_yaml;
+use std::{fs::{self,
+               File},
+          io::Write,
+          path::PathBuf};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LicenseData {
+    pub date_accepted: DateTime<Utc>,
+    pub accepting_product: String,
+    pub accepting_product_version: String,
+    pub user: Option<String>,
```

Comment:
When would we ever want this to be `None`?

Reply:
The `get_current_username()` call that we use to fill this in returns an `Option`, so this needs to accept one as well.  It doesn't matter if it returns `None`.  The metadata is all best effort - none of it is critical.